clustered-indexes.md

```diff
@@ -1,188 +1,225 @@
-|   └─IndexRangeScan_5     | 10.00   | cop[tikv] | table:will_now_cluster, index:b(b) | range:["aaaa","aaaa"], keep order:false, stats:pseudo |
-+--------------------------+---------+-----------+------------------------------------+-------------------------------------------------------+
-3 rows in set (0.00 sec)
+```sql
+mysql> CREATE TABLE t (a VARCHAR(255) PRIMARY KEY CLUSTERED);
+ERROR 8200 (HY000): Cannot create clustered index table when the binlog is ON
 ```
 
-Clustering by a composite `PRIMARY KEY` is also supported:
+If you use clustered indexes together with the attribute `SHARD_ROW_ID_BITS`, TiDB reports the following error:
 
 ```sql
-SET tidb_enable_clustered_index = 1;
-CREATE TABLE composite_primary_key (
- key_a INT NOT NULL,
- key_b INT NOT NULL,
- b CHAR(100),
- PRIMARY KEY (key_a, key_b)
-);
-
-INSERT INTO composite_primary_key VALUES (1, 1, 'aaa'), (2, 2, 'bbb');
-EXPLAIN SELECT * FROM composite_primary_key WHERE key_a = 1 AND key_b = 2;
+mysql> CREATE TABLE t (a VARCHAR(255) PRIMARY KEY CLUSTERED) SHARD_ROW_ID_BITS = 3;
+ERROR 8200 (HY000): Unsupported shard_row_id_bits for table with primary key as row id
 ```
 
-```sql
-Query OK, 0 rows affected (0.00 sec)
+## Compatibility
 
-Query OK, 0 rows affected (0.09 sec)
+### Compatibility with higher and lower TiDB versions
 
-Query OK, 2 rows affected (0.02 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+TiDB supports upgrading tables with clustered indexes but not degrading them, meaning that data in tables with clustered indexes is available on a higher TiDB version, but not on a lower one.
 
-+-------------+---------+------+--------------------------------------------------------------------+---------------+
-| id          | estRows | task | access object                                                      | operator info |
-+-------------+---------+------+--------------------------------------------------------------------+---------------+
-| Point_Get_1 | 1.00    | root | table:composite_primary_key, clustered index:PRIMARY(key_a, key_b) |               |
-+-------------+---------+------+--------------------------------------------------------------------+---------------+
-1 row in set (0.00 sec)
-```
+The clustered indexes feature is partially supported in TiDB v3.0 and v4.0. It is enabled by default when the following requirements are fully met:
+
+- The table contains a `PRIMARY KEY`.
+- The `PRIMARY KEY` consists of only one column.
+- The `PRIMARY KEY` is an `INTEGER`.
 
-This behavior is consistent with MySQL, where the InnoDB storage engine will by default cluster by any `PRIMARY KEY`.
+However, since v5.0, TiDB creates all types of primary keys as non-clustered indexes by default. Because this behavior change may cause TiDB in the default configuration to perform worse in some scenarios, you can consider explicitly specifying clustered indexes.
```

Comment:
```suggestion
However, since v5.0, TiDB creates all types of primary keys as non-clustered indexes by default. If this behavior change might cause TiDB in the default configuration to perform worse in some scenarios, you can consider explicitly specifying using clustered indexes.
```

Reply:
```suggestion
However, since v5.0, TiDB creates all types of primary keys as non-clustered indexes by default. This behavior change might cause TiDB in the default configuration to perform worse in some scenarios. You can consider explicitly specifying a primary key as a clustered index.
```